frontend/packages/console-dynamic-plugin-sdk/src/extensions/feature-flags.ts

```diff
@@ -1,24 +1,16 @@
-import { ExtensionK8sModel } from '../api/common-types';
+import {
+  FeatureFlag as CoreFeatureFlag,
+  ModelFeatureFlag as CoreModelFeatureFlag,
+} from '@openshift/dynamic-plugin-sdk';
 import { Extension, ExtensionDeclaration, CodeRef } from '../types';
 
 /** Gives full control over Console feature flags. */
-export type FeatureFlag = ExtensionDeclaration<
-  'console.flag',
-  {
-    /** Used to set/unset arbitrary feature flags. */
-    handler: CodeRef<FeatureFlagHandler>;
-  }
->;
+export type FeatureFlag = ExtensionDeclaration<'console.flag', CoreFeatureFlag['properties']>;
```

Comment:
We're using the existing Console `ExtensionDeclaration` type instead of another type to avoid adding complexity to the AST parser (`parseExtensionTypeInfo`).

Assuming Console vs. core extension properties are structurally equal, the generated JSON schema for these objects should be structurally equal as well.